File: CleverFerretV2/feature/webfiction/src/main/java/com/cleverferret/v2/feature/webfiction/dto/StoryMetadataDtoV1.java
package com.cleverferret.v2.feature.webfiction.dto;

import com.cleverferret.v2.core.common.VersionedContract;

import java.util.List;

public record StoryMetadataDtoV1(
        String title,
        String author,
        String summary,
        String sourceUrl,
        String sourceSite,
        List<String> tags
) implements VersionedContract {
    @Override
    public String contractVersion() {
        return "v1";
    }
}

File: CleverFerretV2/feature/webfiction/src/main/java/com/cleverferret/v2/feature/webfiction/dto/StoryMetadataDtoV1.kt
package com.cleverferret.v2.feature.webfiction.dto

import com.cleverferret.v2.core.common.VersionedContract
import java.util.List

data class StoryMetadataDtoV1(
    val title: String,
    val author: String,
    val summary: String,
    val sourceUrl: String,
    val sourceSite: String,
    val tags: List<String>
) : VersionedContract {
    override fun contractVersion(): String = "v1"
}
